packages/rules/src/mustSatisfyPeerDependencies.ts

```diff
@@ -71,11 +71,19 @@ export const MATCH_ANY_VERSION_RANGE = /^(\*|x)$/;
  *
  * More info: https://docs.npmjs.com/about-semantic-versioning
  */
-export const MATCH_MAJOR_VERSION_RANGE = /^(\^?\d+|\^?\d+\.x|\^?\d+\.x\.x|\^\d+\.\d+|\^\d+\.\d+\.x|\^\d+\.\d+\.\d+)$/;
+export const MATCH_MAJOR_VERSION_RANGE = /^(\^?\d+|\^?\d+\.x|\^?\d+\.x\.x|\^\d+\.\d+|\^\d+\.\d+\.x|\^\d+\.\d+\.\d+(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)$/;
```

Comment:
some fun stuff here.
mostly just copied from https://regex101.com/r/vkijKf/1/ (referenced in https://semver.org/#spec-item-9)
1. `(?:0|[1-9]\d*)` is a non-capturing group that allows `0` or `[1-9]\d*`. The only difference between this and `\d+` is that it ensures you don't have a leading zero (`0`, `1`, and `10` are allowed, but `01` is not). I didn't bother replacing existing bits of this regex, since this would explode, and I don't see a practical benefit.
1. `(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)` allows numbers without a leading zero, or a leading zero that's followed by at least one alphabetic (or `-`) character, ending in any alphanumeric (or `-`) character.
1. `(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?` is the above, where the second half doesn't require a leading slash for additional segments (e.g. `-rc.1.foo.01`)

You can decompose the build metadata segment the same way

Reply:
i'm definitely just trusting your tests here... can we at least link where you copied this from here as well?

Reply:
sure thing. i made the comments the same for `MATCH_MAJOR_VERSION_RANGE` and `RANGE_REGEX`